backend/core/models.py

```diff
@@ -30,167 +33,273 @@ def create_superuser(self, email, password):
-        default=0.00
+        help_text="The total amount required for the loan.",
     )
     deadline_to_receive_loan = models.DateField(
         help_text="The deadline to receive the loan.",
-        default=timezone.now() + timezone.timedelta(days=365)
+        default=timezone.now() + timezone.timedelta(days=365),
     )
     created_at = models.DateTimeField(
         auto_now_add=True,
-        help_text="The date and time when the loan profile was created."
+        help_text="The date and time when the loan profile was created.",
     )
     updated_at = models.DateTimeField(
         auto_now=True,
-        help_text="The date and time when the loan profile was last updated."
+        help_text="The date and time when the loan profile was last updated.",
     )
     status = models.IntegerField(
         choices=LoanProfileStatus.choices,
         default=LoanProfileStatus.PENDING,
-        help_text="The status of the loan profile."
+        help_text="The status of the loan profile.",
     )
 
+    @property
+    def amount_lended_to_date(self):
+        transaction_total = self.transactions.filter(
+            status=TransactionStatus.COMPLETED
+        ).aggregate(models.Sum("amount"))["amount__sum"]
+
+        return (
+            transaction_total.quantize(Decimal("0.00"))
+            if transaction_total
+            else 0
+        )
+
     class Meta:
         verbose_name = "Loan Profile"
         verbose_name_plural = "Loan Profiles"
-        ordering = ['user']
+        ordering = ["user"]
 
     def __str__(self):
         return f"{self.user.name}'s loan profile"
+
+
+class TransactionManager(models.Manager):
+    """Manager for transactions."""
+
+    def create_transaction(self, lender, borrower, amount, **extra_fields):
+        """Create, save, and return a new transaction."""
+        # user must be User that is lender
+        if lender.role != UserRole.LENDER:
+            raise ValueError("Transaction must be initiated by lender.")
+        # TODO:
+        # borrower must be a loan-profile
+        # must be before the loan-profile cutoff
+        transaction = self.model(
+            user=lender, loan_profile=borrower, amount=amount, **extra_fields
+        )
+        transaction.save(using=self._db)
+
+        return transaction
+
+
+class TransactionStatus(models.IntegerChoices):
+    """Transaction Status choices."""
+
+    PENDING = 1, "Pending"  # Payment initiated but not yet processed
+    COMPLETED = 2, "Completed"  # Payment successfully completed
+    FAILED = 3, "Failed"  # Payment attempt failed
+    REFUNDED = 4, "Refunded"  # Payment was refunded
+    CANCELED = 5, "Canceled"  # Payment was canceled by the user or system
+    ON_HOLD = 6, "On Hold"  # Payment is temporarily on hold
+    CHARGEBACK = 7, "Chargeback"  # Disputed payment
+
+
+class PaymentMethod(models.IntegerChoices):
+    """Payment Method choices."""
+
+    CREDIT_CARD = 1, "Credit Card"
```

Comment:
It might be okay to keep these for future use, but for the MVP PayPal is the only means of payment we need to support.